### DISCUSSION TITLE
handle Api request

### DISCUSSION THREAD

**Discussion by Madhucrj3** [pyodide/pyodide#3799] [Q&A] [answered]
How to use Requests in pyodide 
code:
import requests

response = requests.get("https://jsonplaceholder.typicode.com/todos/1")

if response.status_code == 200:
    data = response.json()
    print(data)
else:
    print("Error:", response.status_code)

//
 I got this error
 Traceback (most recent call last):
OSError: [Errno 50] Protocol not available

During handling of the above exception, another exception occurred:

urllib3.exceptions.NewConnectionError: <urllib3.connection.HTTPSConnection object at 0x1c8e208>: Failed to establish a new connection: [Errno 50] Protocol not available

During handling of the above exception, another exception occurred:

urllib3.exceptions.MaxRetryError: HTTPSConnectionPool(host='jsonplaceholder.typicode.com', port=443): Max retries exceeded with url: /todos/1 (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x1c8e208>: Failed to establish a new connection: [Errno 50] Protocol not available'))

During handling of the above exception, another exception occurred:

requests.exceptions.ConnectionError: HTTPSConnectionPool(host='jsonplaceholder.typicode.com', port=443): Max retries exceeded with url: /todos/1 (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x1c8e208>: Failed to establish a new connection: [Errno 50] Protocol not available'))

how to resolve above error ??

**Comment by ryanking13** [comment, marked as answer]
See:

- https://pyodide.org/en/stable/usage/faq.html#how-can-i-load-external-files-in-pyodide
- https://github.com/koenvo/pyodide-http
